src/Client/PullPayment.php

```diff
@@ -0,0 +1,336 @@
+    | Approve, cancel or mark a payout as paid.
+    |
+    */
+
+    /**
+    * Get Store's Pull Payments
+    *
+    * @return string $id
+    * @return string $name
+    * @return string $currency
+    * @return int $amount
+    * @return int $period
+    * @return int $BOLT11Expiration
+    * @return bool $archived
+    * @return string $viewlink
+    */
+
+    public function getStorePullPayments(
+        string $storeId,
+        bool $includeArchived
+    ) {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $body = json_encode(
+            [
+                'includeArchived' => $includeArchived,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+    * Create a Pull Payment
+    *
+    * @return string $id
+    * @return string $name
+    * @return string $currency
+    * @return int $amount
+    * @return int $period
+    * @return int $BOLT11Expiration
+    * @return bool $archived
+    * @return string $viewlink
+    */
+
+    public function createPullPayment(
+        string $storeId,
+        string $paymentName,
+        string $paymentAmount,
+        string $paymentCurrency,
+        mixed $paymentPeriod,
+        mixed $boltExpiration,
+        mixed $startsAt,
+        mixed $expiresAt,
+        array $paymentMethods
+    ) {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'POST';
+
+        $body = json_encode(
+            [
+                'name' => $paymentName,
+                'amount' => $paymentAmount,
+                'currency' => $paymentCurrency,
+                'period' => $paymentPeriod,
+                'BOLT11Expiration' => $boltExpiration,
+                'startsAt' => $startsAt,
+                'expiresAt' => $expiresAt,
+                'paymentMethods' => $paymentMethods
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
+        } else {
+            throw $this->getExceptionByStatusCode($method, $url, $response);
+        }
+    }
+
+    /**
+    * Archive a Pull Payment
+    *
+    * @return true on success, throws exception
+    */
+
+    public function archivePullPayment(
+        string $storeId,
+        string $pullPaymentId
+    ) {
```

Comment:
you can avoid the docblock and add return type hint `: bool` here, see other classes for examples